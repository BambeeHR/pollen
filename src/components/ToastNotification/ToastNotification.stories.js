import { boolean, select, text } from '@storybook/addon-knobs';
import Notification from '../../constants/Notification';
import ToastNotification from './ToastNotification.vue';
import BaseButton from '../BaseButton/BaseButton.vue';
import TypeOverline from '../TypeOverline/TypeOverline.vue';

const { Variants } = Notification;

export default {
  title: 'ToastNotification',
  component: ToastNotification,
};

export const Gallery = () => ({
  components: { ToastNotification, TypeOverline },
  data() {
    return {
      variants: Object.values(Variants),
    };
  },
  template: `
  <div style="width: 400px">
    <div v-for="variant in variants" :key="variant" class="mb-4">
      <TypeOverline tag="h2" class="mb-2">{{ variant }}</TypeOverline>
      <ToastNotification
        message="Lorem ipsum dolor sit omet"
        class="mb-2"
        :variant="variant"
      />
      <ToastNotification
        message="Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat."
        class="mb-2"
        :variant="variant"
      />
      <ToastNotification
        message="Lorem ipsum dolor sit omet"
        class="mb-2"
        image="https://cdn.filestackcontent.com/enCa5BTVeIsLHcgkdpAc"
        :variant="variant"
      />
      <ToastNotification
      message="Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat."
        class="mb-2"
        initials="XO"
        :variant="variant"
      />
    </div>
  </div>
  `,
});

export const WithMessageAsASlot = () => ({
  components: { ToastNotification },
  template: `
  <ToastNotification
    image="https://cdn.filestackcontent.com/enCa5BTVeIsLHcgkdpAc"
  >
    Ut <a href="#" class="text-white">enim ad minim</a> veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
  </ToastNotification>
  `,
});

export const NoActions = () => ({
  components: { ToastNotification },
  template: `
  <ToastNotification
    message="Lorem ipsum dolor sit omet"
    :showActions="false"
  />
  `,
});

export const WithActionAsSlot = () => ({
  components: { ToastNotification, BaseButton },
  template: `
  <ToastNotification
    image="https://cdn.filestackcontent.com/enCa5BTVeIsLHcgkdpAc"
    message="Lorem ipsum dolor sit omet"
  >
    <template #actions>
      <BaseButton flat variant="inverted" size="small" @click="$emit('confirm')">Confirm</BaseButton>
      <BaseButton flat variant="inverted" size="small" @click="$emit('cancel')">Undo</BaseButton>
    </template>
  </ToastNotification>
  `,
});

export const WithKnobs = () => ({
  components: { ToastNotification },
  props: {
    image: {
      default: text(
        'Image',
        'https://cdn.filestackcontent.com/enCa5BTVeIsLHcgkdpAc'
      ),
    },
    initials: {
      default: text('Initials', 'XO'),
    },
    message: {
      default: text('Message', 'Lorem ipsum dolor sit omet'),
    },
    showActions: {
      default: boolean('Show Actions', true),
    },
    variant: {
      default: select('Variant', Object.values(Variants), Variants.DEFAULT),
    },
  },
  template: `
  <ToastNotification
    :image="image"
    :initials="initials"
    :message="message"
    :showActions="showActions"
    :variant="variant"
  />
  `,
});
