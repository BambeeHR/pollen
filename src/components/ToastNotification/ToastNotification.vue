<template>
  <div
    class="toast-notification"
    :class="[
      `toast-notification--${variant}`,
      { 'toast-notification--with-avatar': !!avatar },
    ]"
    :role="showActions ? 'alertdialog' : 'alert'"
  >
    <div class="toast-notification__content">
      <div v-if="avatar" class="toast-notification__avatar-container">
        <AvatarImage v-bind="avatar" class="toast-notification__avatar" />
      </div>
      <div class="toast-notification__message">
        <slot>{{ message }}</slot>
      </div>
      <div v-if="showActions" class="toast-notification__actions">
        <slot name="actions">
          <IconButton
            v-bind="actionProps"
            title="Close"
            class="toast-notification__close"
            @click="$emit('close')"
          />
        </slot>
      </div>
    </div>
  </div>
</template>

<script>
import AvatarImage from '../AvatarImage/AvatarImage.vue';
import Button from '../../constants/Button';
import IconButton from '../IconButton/IconButton.vue';
import Notification from '../../constants/Notification';
import { Icons } from '../BaseIcon/BaseIcon.vue';

/**
 * A notification component. Designed to displayed within a `ToastNotifications`
 * container that would display this in a fixed, overlayed position. Also
 * referred to as a "Snackbar" or "Meerkat".
 */
export default {
  components: { AvatarImage, IconButton },
  props: {
    /** The url of an avatar image to display. */
    image: {
      type: String,
      default: null,
    },
    /**
     * A fallback pair of initials to display if no image is supplied, but a
     * prepended avatar component is still desired.
     */
    initials: {
      type: String,
      default: null,
    },
    /**
     * The alert message. This can be passed in as a prop, or as the
     * default slot.
     */
    message: {
      type: String,
      default: null,
    },
    /**
     * If true, displays by default an `x` button that emits a `close` event.
     * This button can be overridden by using the `#actions` named slot.
     */
    showActions: {
      type: Boolean,
      default: true,
    },
    /** One of `default`, `secondary`, or `error`. */
    variant: {
      type: String,
      default: Notification.Variants.DEFAULT,
      validator: (value) =>
        Object.values(Notification.Variants).includes(value),
    },
  },
  computed: {
    actionProps() {
      return {
        size: Button.Sizes.MEDIUM,
        icon: Icons.CLOSE,
        variant: Button.Variants.INVERTED,
        title: 'Close',
        flat: true,
      };
    },
    avatar() {
      const { image, initials } = this;
      if (image || initials) {
        return { image, initials };
      }
      return null;
    },
  },
};
</script>

<style scoped>
.toast-notification__content {
  @apply box-border flex font-body-small items-center relative rounded-md shadow-6 text-white py-3 px-4;
  min-height: 3rem;
}

.toast-notification__avatar-container {
  @apply absolute left-0 -ml-6 p-1 rounded-full top-0;
}

.toast-notification--default .toast-notification__content,
.toast-notification--default .toast-notification__avatar-container {
  @apply bg-gray-0;
}

.toast-notification--seconary .toast-notification__content,
.toast-notification--seconary .toast-notification__avatar-container {
  @apply bg-secondary;
}

.toast-notification--error .toast-notification__content,
.toast-notification--error .toast-notification__avatar-container {
  @apply bg-error;
}

.toast-notification__message {
  @apply flex-grow;
}

.toast-notification__actions {
  @apply flex-shrink-0 flex items-center ml-3 -mr-1 -my-1 self-stretch;
}

.toast-notification .icon-button {
  @apply opacity-50;
}

.toast-notification--with-avatar {
  @apply pl-6;
}

.toast-notification--with-avatar .toast-notification__content {
  @apply pl-8;
}
</style>
