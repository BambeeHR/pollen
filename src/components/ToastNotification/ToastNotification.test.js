import { mount } from '@vue/test-utils';
import ToastNotification from './ToastNotification.vue';
import { Gallery } from './ToastNotification.stories';

describe('ToastNotification', () => {
  test('renders correctly', () => {
    const wrapper = mount(ToastNotification);
    expect(wrapper.element).toMatchSnapshot();
  });

  test('renders gallery correctly', () => {
    const gallery = Gallery();
    const wrapper = mount(gallery);
    expect(wrapper.element).toMatchSnapshot();
  });

  test('emits a `close` event', () => {
    const onClose = jest.fn();
    const wrapper = mount(ToastNotification, {
      listeners: {
        close: onClose,
      },
    });
    wrapper.find('.toast-notification__close').trigger('click');
    expect(onClose).toHaveBeenCalled();
  });
});
