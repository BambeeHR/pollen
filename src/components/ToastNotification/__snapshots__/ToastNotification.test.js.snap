// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastNotification renders correctly 1`] = `
<div
  class="toast-notification toast-notification--default"
  role="alertdialog"
>
  <div
    class="toast-notification__content"
  >
    <!---->
     
    <div
      class="toast-notification__message"
    >
      
    </div>
     
    <div
      class="toast-notification__actions"
    >
      <button
        class="toast-notification__close toast-notification__close icon-button icon-button--inverted icon-button--medium icon-button--flat"
        title="Close"
      >
        <svg
          aria-hidden="true"
          class="base-icon svg-inline--fa fa-times fa-w-10 icon-button__icon"
          data-icon="times"
          data-prefix="fal"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class=""
            d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
            fill="currentColor"
          />
        </svg>
        <span
          class="icon-button__title"
        >
          Close
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`ToastNotification renders gallery correctly 1`] = `
<div
  style="width: 400px;"
>
  <div
    class="mb-4"
  >
    <h2
      class="type-overline type-overline-small mb-2"
    >
      default
    </h2>
     
    <div
      class="toast-notification mb-2 toast-notification--default"
      role="alertdialog"
    >
      <div
        class="toast-notification__content"
      >
        <!---->
         
        <div
          class="toast-notification__message"
        >
          Lorem ipsum dolor sit omet
        </div>
         
        <div
          class="toast-notification__actions"
        >
          <button
            class="toast-notification__close toast-notification__close icon-button icon-button--inverted icon-button--medium icon-button--flat"
            title="Close"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 icon-button__icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
            <span
              class="icon-button__title"
            >
              Close
            </span>
          </button>
        </div>
      </div>
    </div>
     
    <div
      class="toast-notification mb-2 toast-notification--default"
      role="alertdialog"
    >
      <div
        class="toast-notification__content"
      >
        <!---->
         
        <div
          class="toast-notification__message"
        >
          Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </div>
         
        <div
          class="toast-notification__actions"
        >
          <button
            class="toast-notification__close toast-notification__close icon-button icon-button--inverted icon-button--medium icon-button--flat"
            title="Close"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 icon-button__icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
            <span
              class="icon-button__title"
            >
              Close
            </span>
          </button>
        </div>
      </div>
    </div>
     
    <div
      class="toast-notification mb-2 toast-notification--default toast-notification--with-avatar"
      role="alertdialog"
    >
      <div
        class="toast-notification__content"
      >
        <div
          class="toast-notification__avatar-container"
        >
          <div
            class="avatar-image avatar-image--rounded toast-notification__avatar"
            style="background-image: url(https://cdn.filestackcontent.com/resize=width:40,height:40/enCa5BTVeIsLHcgkdpAc); font-size: 1.0625rem; height: 2.5rem; width: 2.5rem;"
          >
            <!---->
          </div>
        </div>
         
        <div
          class="toast-notification__message"
        >
          Lorem ipsum dolor sit omet
        </div>
         
        <div
          class="toast-notification__actions"
        >
          <button
            class="toast-notification__close toast-notification__close icon-button icon-button--inverted icon-button--medium icon-button--flat"
            title="Close"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 icon-button__icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
            <span
              class="icon-button__title"
            >
              Close
            </span>
          </button>
        </div>
      </div>
    </div>
     
    <div
      class="toast-notification mb-2 toast-notification--default toast-notification--with-avatar"
      role="alertdialog"
    >
      <div
        class="toast-notification__content"
      >
        <div
          class="toast-notification__avatar-container"
        >
          <div
            class="avatar-image avatar-image--rounded toast-notification__avatar"
            style="font-size: 1.0625rem; height: 2.5rem; width: 2.5rem;"
          >
            <div
              class="avatar-image__initials"
            >
              
    XO
  
            </div>
          </div>
        </div>
         
        <div
          class="toast-notification__message"
        >
          Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </div>
         
        <div
          class="toast-notification__actions"
        >
          <button
            class="toast-notification__close toast-notification__close icon-button icon-button--inverted icon-button--medium icon-button--flat"
            title="Close"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 icon-button__icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
            <span
              class="icon-button__title"
            >
              Close
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mb-4"
  >
    <h2
      class="type-overline type-overline-small mb-2"
    >
      seconary
    </h2>
     
    <div
      class="toast-notification mb-2 toast-notification--seconary"
      role="alertdialog"
    >
      <div
        class="toast-notification__content"
      >
        <!---->
         
        <div
          class="toast-notification__message"
        >
          Lorem ipsum dolor sit omet
        </div>
         
        <div
          class="toast-notification__actions"
        >
          <button
            class="toast-notification__close toast-notification__close icon-button icon-button--inverted icon-button--medium icon-button--flat"
            title="Close"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 icon-button__icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
            <span
              class="icon-button__title"
            >
              Close
            </span>
          </button>
        </div>
      </div>
    </div>
     
    <div
      class="toast-notification mb-2 toast-notification--seconary"
      role="alertdialog"
    >
      <div
        class="toast-notification__content"
      >
        <!---->
         
        <div
          class="toast-notification__message"
        >
          Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </div>
         
        <div
          class="toast-notification__actions"
        >
          <button
            class="toast-notification__close toast-notification__close icon-button icon-button--inverted icon-button--medium icon-button--flat"
            title="Close"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 icon-button__icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
            <span
              class="icon-button__title"
            >
              Close
            </span>
          </button>
        </div>
      </div>
    </div>
     
    <div
      class="toast-notification mb-2 toast-notification--seconary toast-notification--with-avatar"
      role="alertdialog"
    >
      <div
        class="toast-notification__content"
      >
        <div
          class="toast-notification__avatar-container"
        >
          <div
            class="avatar-image avatar-image--rounded toast-notification__avatar"
            style="background-image: url(https://cdn.filestackcontent.com/resize=width:40,height:40/enCa5BTVeIsLHcgkdpAc); font-size: 1.0625rem; height: 2.5rem; width: 2.5rem;"
          >
            <!---->
          </div>
        </div>
         
        <div
          class="toast-notification__message"
        >
          Lorem ipsum dolor sit omet
        </div>
         
        <div
          class="toast-notification__actions"
        >
          <button
            class="toast-notification__close toast-notification__close icon-button icon-button--inverted icon-button--medium icon-button--flat"
            title="Close"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 icon-button__icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
            <span
              class="icon-button__title"
            >
              Close
            </span>
          </button>
        </div>
      </div>
    </div>
     
    <div
      class="toast-notification mb-2 toast-notification--seconary toast-notification--with-avatar"
      role="alertdialog"
    >
      <div
        class="toast-notification__content"
      >
        <div
          class="toast-notification__avatar-container"
        >
          <div
            class="avatar-image avatar-image--rounded toast-notification__avatar"
            style="font-size: 1.0625rem; height: 2.5rem; width: 2.5rem;"
          >
            <div
              class="avatar-image__initials"
            >
              
    XO
  
            </div>
          </div>
        </div>
         
        <div
          class="toast-notification__message"
        >
          Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </div>
         
        <div
          class="toast-notification__actions"
        >
          <button
            class="toast-notification__close toast-notification__close icon-button icon-button--inverted icon-button--medium icon-button--flat"
            title="Close"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 icon-button__icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
            <span
              class="icon-button__title"
            >
              Close
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mb-4"
  >
    <h2
      class="type-overline type-overline-small mb-2"
    >
      error
    </h2>
     
    <div
      class="toast-notification mb-2 toast-notification--error"
      role="alertdialog"
    >
      <div
        class="toast-notification__content"
      >
        <!---->
         
        <div
          class="toast-notification__message"
        >
          Lorem ipsum dolor sit omet
        </div>
         
        <div
          class="toast-notification__actions"
        >
          <button
            class="toast-notification__close toast-notification__close icon-button icon-button--inverted icon-button--medium icon-button--flat"
            title="Close"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 icon-button__icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
            <span
              class="icon-button__title"
            >
              Close
            </span>
          </button>
        </div>
      </div>
    </div>
     
    <div
      class="toast-notification mb-2 toast-notification--error"
      role="alertdialog"
    >
      <div
        class="toast-notification__content"
      >
        <!---->
         
        <div
          class="toast-notification__message"
        >
          Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </div>
         
        <div
          class="toast-notification__actions"
        >
          <button
            class="toast-notification__close toast-notification__close icon-button icon-button--inverted icon-button--medium icon-button--flat"
            title="Close"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 icon-button__icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
            <span
              class="icon-button__title"
            >
              Close
            </span>
          </button>
        </div>
      </div>
    </div>
     
    <div
      class="toast-notification mb-2 toast-notification--error toast-notification--with-avatar"
      role="alertdialog"
    >
      <div
        class="toast-notification__content"
      >
        <div
          class="toast-notification__avatar-container"
        >
          <div
            class="avatar-image avatar-image--rounded toast-notification__avatar"
            style="background-image: url(https://cdn.filestackcontent.com/resize=width:40,height:40/enCa5BTVeIsLHcgkdpAc); font-size: 1.0625rem; height: 2.5rem; width: 2.5rem;"
          >
            <!---->
          </div>
        </div>
         
        <div
          class="toast-notification__message"
        >
          Lorem ipsum dolor sit omet
        </div>
         
        <div
          class="toast-notification__actions"
        >
          <button
            class="toast-notification__close toast-notification__close icon-button icon-button--inverted icon-button--medium icon-button--flat"
            title="Close"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 icon-button__icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
            <span
              class="icon-button__title"
            >
              Close
            </span>
          </button>
        </div>
      </div>
    </div>
     
    <div
      class="toast-notification mb-2 toast-notification--error toast-notification--with-avatar"
      role="alertdialog"
    >
      <div
        class="toast-notification__content"
      >
        <div
          class="toast-notification__avatar-container"
        >
          <div
            class="avatar-image avatar-image--rounded toast-notification__avatar"
            style="font-size: 1.0625rem; height: 2.5rem; width: 2.5rem;"
          >
            <div
              class="avatar-image__initials"
            >
              
    XO
  
            </div>
          </div>
        </div>
         
        <div
          class="toast-notification__message"
        >
          Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </div>
         
        <div
          class="toast-notification__actions"
        >
          <button
            class="toast-notification__close toast-notification__close icon-button icon-button--inverted icon-button--medium icon-button--flat"
            title="Close"
          >
            <svg
              aria-hidden="true"
              class="base-icon svg-inline--fa fa-times fa-w-10 icon-button__icon"
              data-icon="times"
              data-prefix="fal"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class=""
                d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
                fill="currentColor"
              />
            </svg>
            <span
              class="icon-button__title"
            >
              Close
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
